pkg/dockerfile/dockerfile.go

```diff
@@ -37,6 +42,25 @@ func Parse(b []byte) ([]instructions.Stage, error) {
 	return stages, err
 }
 
+// ResolveStages resolves any calls to previous stages with names to indices
+// Ex. --from=second_stage should be --from=1 for easier processing later on
+func ResolveStages(stages []instructions.Stage) {
+	nameToIndex := make(map[string]string)
+	for i, stage := range stages {
+		index := strconv.Itoa(i)
+		nameToIndex[stage.Name] = index
```

Comment:
it looks like the second one here overwrites the first one. Do we need both?

Reply:
so if the stage has a name associated the second won't override the first, so I added both in just to make it easier to resolve '--from' from names to indexes later on

Reply:
Ah ok. Maybe it would be more clear to have two separate maps?

Reply:
I guess it would get confusing if someone tried to name a stage with an index...

Reply:
Oh I see, that makes sense --  I changed it a bit to be less confusing with one map